packages/jsapi-components/src/spectrum/Picker/Picker.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+  NormalizedPickerItemData,
+  Picker as PickerBase,
+  PickerProps as PickerPropsBase,
+} from '@deephaven/components';
+import { dh } from '@deephaven/jsapi-types';
+import { PICKER_ITEM_HEIGHT } from '@deephaven/utils';
+import { useEffect } from 'react';
+import { useViewportData } from '../../useViewportData';
+import { getPickerKeyColumn } from './PickerUtils';
+import usePickerItemRowDeserializer from './usePickerItemRowDeserializer';
+
+export interface PickerProps extends PickerPropsBase {
+  table: dh.Table;
+  /* The column of values to use as item keys. Defaults to the first column. */
+  keyColumn?: string;
+  /* The column of values to display as primary text. Defaults to the `keyColumn` value. */
+  labelColumn?: string;
+}
+
+export function Picker({
+  table,
+  keyColumn: keyColumnName,
+  labelColumn: labelColumnName,
+  selectedKey,
+  ...props
+}: PickerProps): JSX.Element {
+  const deserializeRow = usePickerItemRowDeserializer({
+    table,
+    keyColumnName,
+    labelColumnName,
+  });
+
+  const { viewportData, onScroll, setViewport } = useViewportData<
+    NormalizedPickerItemData,
+    dh.Table
+  >({
+    reuseItemsOnTableResize: true,
+    table,
+    itemHeight: PICKER_ITEM_HEIGHT,
+    deserializeRow,
+  });
+
+  useEffect(
+    // Set viewport to include the selected item so that its data will load and
+    // the real `key` will be available to show the selection in the UI.
+    function setViewportFromSelectedKey() {
+      if (selectedKey == null) {
+        return;
+      }
+
+      const keyColumn = getPickerKeyColumn(table, keyColumnName);
+
+      table
+        .seekRow(0, keyColumn, keyColumn.type, selectedKey)
```

Comment:
Looks like passing `keyColumn.type` directly here won't always work. I'm getting a `java.lang.UnsupportedOperationException: Invalid value type for seekRow: java.lang.String` error in certain scenarios. Will have to see about mapping this to expected type.